src/System.Device.Gpio.Tests/RaspberryPiDriverTests.cs

```diff
@@ -27,6 +28,20 @@ public RaspberryPiDriverTests(ITestOutputHelper testOutputHelper)
 
         protected override PinNumberingScheme GetTestNumberingScheme() => PinNumberingScheme.Logical;
 
+        private bool IsRaspi4()
+        {
+            if (File.Exists("/proc/device-tree/model"))
```

Comment:
I'd prefer we used APIs exposed in System.Device for this, if not available at least check for type name for the driver - I do not want to duplicate logic here. I feel this might become an issue once we Raspi5/Raspi6 shows up or some other board shows up. Even better you can [check if InputPullDown mode is supported](https://docs.microsoft.com/en-us/dotnet/api/system.device.gpio.gpiocontroller.ispinmodesupported?view=iot-dotnet-1.3#System_Device_Gpio_GpioController_IsPinModeSupported_System_Int32_System_Device_Gpio_PinMode_)

Reply:
We do not expose this API, therefore it needs duplication here. I'm also not very happy about this. We have the following possibilities:

- Use InternalsVisibleTo and change the implementation in System.Device to internal.
- Make the implementation public
- Make the implementation public, but with a more generic API.

I would prefer 3, that means having some generic hardware detection API. But this is way beyond this PR.

Reply:
I guess we can fix this when it becomes a problem